tests/parametric_study/test_parametric_study.py

```diff
@@ -1736,3 +1737,163 @@ def test_set_iteration_sets_iteration(tmp_path: pytest.TempPathFactory):
+        pathlib.Path("csv") / "microstructure-test-study.csv"
+    )
+    # act, assert
+    study = ps.ParametricStudy(tmp_path / study_name)
+    study.import_csv_study(single_bead_demo_csv_file)
+    study.import_csv_study(porosity_demo_csv_file)
+    study.import_csv_study(microstructure_demo_csv_file)
+    assert len(study.data_frame()) == 15
+
+
+def test_import_csv_study_adds_simulations_to_existing_study(tmp_path: pytest.TempPathFactory):
+    # arrange
+    study_name = "test_study"
+    single_bead_demo_csv_file = test_utils.get_test_file_path(
+        pathlib.Path("csv") / "single-bead-test-study.csv"
+    )
+    # act, assert
+    study = ps.ParametricStudy(tmp_path / study_name)
+    powers = [50, 250, 700]
+    scan_speeds = [0.35]
+
+    # act
+    study.generate_porosity_permutations(
+        material_name="material",
+        laser_powers=powers,
+        scan_speeds=scan_speeds,
+    )
+
+    # act
+    study.import_csv_study(single_bead_demo_csv_file)
+    assert len(study.data_frame()) == 8
+
+
+@pytest.mark.parametrize(
+    "file_name, argument",
+    [
+        ("completed-test-study.csv", True),
+        ("pending-test-study.csv", False),
+        ("skip-test-study.csv", False),
+    ],
+)
+def test_import_csv_study_calls_remove_duplicates_entries_correctly(
+    monkeypatch, file_name, argument, tmp_path: pytest.TempPathFactory
+):
+    study = ps.ParametricStudy(tmp_path / "test_study")
+    csv_file = test_utils.get_test_file_path(pathlib.Path("csv") / file_name)
+    patched_simulate = create_autospec(
+        ps.ParametricStudy._remove_duplicate_entries, return_value=[]
+    )
+    monkeypatch.setattr(ps.ParametricStudy, "_remove_duplicate_entries", patched_simulate)
+
+    # act
+    study.import_csv_study(csv_file)
+
+    # assert
+    assert patched_simulate.call_args[1]["overwrite"] == argument
+
+
+def test_import_csv_study_drops_duplicates_with_correct_simulation_status_heirarchy(
+    tmp_path: pytest.TempPathFactory,
+):
+    # arrange
+    study_name = "test_study"
+    duplicate_rows_file = test_utils.get_test_file_path(pathlib.Path("csv") / "duplicate-rows.csv")
+    # act, assert
+    study = ps.ParametricStudy(tmp_path / study_name)
+    study.import_csv_study(duplicate_rows_file)
+    df = study.data_frame()
+    assert len(df) == 3
+    assert df.iloc[0][ps.ColumnNames.STATUS] == SimulationStatus.COMPLETED
+    assert df.iloc[0][ps.ColumnNames.ID] == "sb_c_d"
+    assert df.iloc[1][ps.ColumnNames.STATUS] == SimulationStatus.PENDING
+    assert df.iloc[1][ps.ColumnNames.ID] == "sb_p"
+    assert df.iloc[2][ps.ColumnNames.STATUS] == SimulationStatus.SKIP
+    assert df.iloc[2][ps.ColumnNames.ID] == "sb_s"
+
+
+@pytest.mark.parametrize(
+    "index, error_message",
+    [
+        (0, "Invalid parameter combination"),
+        (1, "Simulations with status Error will not be added to the parametric study"),
+        (2, "Invalid simulation input type: invalid"),
```

Comment:
Hmmm. This message is not very useful. What's the condition that caused it to be invalid and can we provide a more detailed message?